cmd/agent/subcommands/flare/command_windows_test.go

```diff
@@ -32,7 +33,18 @@ func sysprobeSocketPath(_ *testing.T) string {
 func NewSystemProbeTestServer(handler http.Handler) (*httptest.Server, error) {
 	server := httptest.NewUnstartedServer(handler)
 
-	conn, err := sysprobeserver.NewListener(systemProbeTestPipeName)
+	// Prepare a security descriptor that allows the current user.
+	currentUser, err := user.Current()
+	if err != nil {
+		return nil, err
+	}
+
+	sd, err := sysprobeserver.FormatSecurityDescriptorWithSid(currentUser.Uid)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := sysprobeserver.NewListenerWithSecurityDescriptor(systemProbeTestPipeName, sd)
```

Comment:
I am leaning to "no" because:
1. The security descriptor configures other things besides a specific user.
2. I think NewListenerForCurrentUser specifically for tests-only should be fine.  This API is not meant for general use, should not be widely shared, because it lends to misuse/abuse/naive use.